libyul/backends/evm/EVMCodeTransform.cpp

```diff
@@ -239,9 +239,12 @@ void CodeTransform::operator()(FunctionCall const& _call)
 
 	m_assembly.setSourceLocation(extractSourceLocationFromDebugData(_call.debugData));
 	if (BuiltinFunctionForEVM const* builtin = m_dialect.builtin(_call.functionName.name))
-		builtin->generateCode(_call, m_assembly, m_builtinContext, [&](Expression const& _expression) {
-			visitExpression(_expression);
-		});
+	{
+		for (auto&& [i, arg]: _call.arguments | ranges::views::enumerate | ranges::views::reverse)
+			if (!builtin->literalArgument(i))
+				visitExpression(arg);
+		builtin->generateCode(_call, m_assembly, m_builtinContext);
```

Comment:
reset source location?

Reply:
The builtin code looked like it's meant to take care of source locations - but it doesn't hurt to also set it here again as well, so I'll add it in any case. We could consider removing that part from ``generateCode`` then, but we can also keep it on both sides.

Reply:
Added